### ISSUE TITLE
I need some clarifications

### ISSUE THREAD

**Issue by yasser-sobhy** [uNetworking/uWebSockets#700]
Hi, I am planning to do two things, create a tutorial for uWebSockets and  write a pub/sub based on it. I have read the source code yesterday and will review it today again, but I still need some clarifications about the uWebSockets:

1- Group:
I noticed that `Group` is used internally by `Hub`, but can also be created manually using `Hub.createGroup()`. So can `Group` be used in pub/sub service? I mean if I create a `Group` manually and add websockets to it, can it be used to group websockets like a channel? can I use `Group::broadcast()` to broadcast a message to all websockets inside a Group? Can a websocket be used inside more than one `Group`?

```
    uWS::Hub h;
// MyGroup inherits uWS::Group, so I can call  group.addSocket(ws)
MyGroup<uWS::SERVER> *group = h.createGroup<uWS::SERVER>(uWS::PERMESSAGE_DEFLATE);

    h.onConnection([&h, &connections, broadcastMessage, broadcastMessageLength](uWS::WebSocket<uWS::SERVER> *ws, uWS::HttpRequest req) {
     group.addSocket(ws)
    });
```

2- Threading
In a pub/sub service multi-threading is an essential part, for example to implement channels I may create many `Group`s, each representing a channel and every channel has it's own thread. A websocket may be subscribed to many channels at the same time. I know that `Group::broadcast()` && `Websocket` support multi-threading, so here I am asking about the design, do you have something that many not make the above scenario work? Something implemented internally by uWebSockets or do you have an advice or recommendations before starting?

3- Client Side Support
For my pub/sub service, I will use uWebSockets for server-side websocket handling, but I will write about client side support in the tutorial. So what platforms uWebSockets supports for client side other than those supported by server-side? Do you have advices or recommendations about this?

4- Compression
I have seen some issues asking for compression support, and it was said that compression is not a priority, but in `WebSocket` I have seen compression support. So I need more information about compression support.

**Comment by alexhultman** [comment]
1. https://github.com/uNetworking/uWebSockets/issues/693
2: Don't use threading like that, follow the multithreaded example (1 thread per hub, no data dependencies). Or simply don't use threading at all, you don't need it in most cases.
3: Same but client side doesn't matter (it's supposed to be a web browser).
4. Not fully supported yet, will probably add it sometime when I have the time / interest.

Other than this, you know pub/sub is easy to make an absolute pile of garbage of. Most if not all implementations of pub/sub is complete shit and completely misses the point of what it is trying to achieve in the first place.

You cannot simply have a for loop with broadcast in it, then you are a moron and are not going to get any better results than any shitty socket.io crap. This is exactly why I started µTT to show this because I got fed up with all these shitty benchmarks made by complete morons (self-proclaimed benchmark professionals, etc) which basically implement pub/sub with µWS only to come to the conclusion µWS is no better than some shitty PHP implementation. If you aim with your asshole then a sniper rifle is no better than a shotgun at long range you know.

Also I think the world has seen too many pub/sub implementations already. Make a snow angel instead or something.
